auth/workload/provider.go

```diff
@@ -29,6 +29,9 @@ type IdentityProviderConfig struct {
 	// File sources the subject credential from a file.
 	File *FileCredentialSource `json:"file,omitempty"`
 
+	// Token sources the credentials from a directly supplied token.
+	Token *CredentialTokenSource `json:"token,omitempty"`
```

Comment:
```suggestion
	Token *CredentialTokenSource `json:"-"`
```

If we do this the token will never be written to a file, and it won't be possible to load it. Maybe we want to do this to discourage users from saving credentials to their own filesystems?

Reply:
Let's leave it with the JSON tag. There maybe a use case for being able to pass a cred file as a string with the token embedded or if rendering the cred file directly it could save from having to write two files.